textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexMatcher.java

```diff
@@ -34,22 +52,58 @@ public void open() throws DataFlowException {
 
     @Override
     public ITuple getNextTuple() throws DataFlowException {
-
-        try {
+		try {
             ITuple sourceTuple = sourceOperator.getNextTuple();
             if(sourceTuple == null){
                 return null;
             }
-            if (predicate.satisfy(sourceTuple)) {
-                return sourceTuple;
-            } else {
-                return getNextTuple();
+            
+            RegexPredicate regexPredicate = (RegexPredicate)predicate; 
+            
+            spans = regexPredicate.statisfySpan(sourceTuple);
+            
+            if (spans.size() != 0) { // a list of matches found
+            	if (schema == null || spanSchema == null) {
+            		schema = sourceTuple.getSchema();
+            		spanSchema = createSpanSchema();
+            	}
+            	fields = sourceTuple.getFields();
+            	return getSpanTuple(spans);
+            } else { // no match found
+            	return getNextTuple();
             }
         } catch (Exception e) {
             e.printStackTrace();
             throw new DataFlowException(e.getMessage(), e);
-        }
-
+        }        
+    }
+    
+    /**
+     * This function converts the schema of original tuples to the schema of match results returned by RegexMatcher. 
+     * It adds SPAN_LIST_ATTRIBUTE in the end of the schema of tuples from lower level.
+     * Because we need to use span to specify the position of matched sequence in this tuple.
+     * For example, if a tuple ("george watson", "staff", 33, "(949)888-8888") matched "g[^\s]*",
+     * RegexMatcher returns ("george watson", "staff", 33, "(949)888-8888", [Span(name, 0, 6, "g[^\s]*", "george watson")])
+     * 
+     * @return a schema object = attributes in original tuples followed by SPAN_LIST_ATTRIBUTE
+     */
+    private Schema createSpanSchema() {
```

Comment:
@sandeepreddy602 wrote a createSchema function in another PR.

Check
https://github.com/TextDB/textdb/pull/85/files/f5f58f61ac69d6fa9f727758fea452439f3dc25c#r61792027

@prakul : Regex team is waiting for your PR to merge as well.  

@laisycs : please wait for the merge of "Keyword Operator" then reuse that function.